src/common/types/interval_t.cpp

```diff
@@ -97,50 +102,229 @@ void Interval::addition(interval_t& result, uint64_t number, std::string specifi
     }
 }
 
-void Interval::parseIntervalField(
-    std::string buf, uint64_t& pos, uint64_t len, interval_t& result) {
-    uint64_t number;
-    uint64_t offset = 0;
-    // parse digits
-    number = std::stoi(buf.c_str() + pos, reinterpret_cast<size_t*>(&offset));
-    pos += offset;
-    // skip spaces
-    while (pos < len && isspace(buf[pos])) {
-        pos++;
+template<class T>
+T IntervalTryCastInteger(int64_t input) {
```

Comment:
We always use camel case for function/var names, so i would rename it to intervalTryCastInteger